fix(tests): free up space in some test runners

droctothorpe opened the pull request:
**Description of your changes:**

This PR follows a suggestion documented [here](https://github.com/actions/runner-images/issues/2840#issuecomment-1284059930) to free up 8GB of space on `/dev/root` on the sdk execution test runners. This is intended to address disk space issues that have surfaced intermittently during CI.

**Checklist:**
- [x] You have [signed off your commits](https://www.kubeflow.org/docs/about/contributing/#sign-off-your-commits)
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 
<!--
   PR titles examples:
    * `fix(frontend): fixes empty page. Fixes #1234`
       Use `fix` to indicate that this PR fixes a bug.
    * `feat(backend): configurable service account. Fixes #1234, fixes #1235`
       Use `feat` to indicate that this PR adds a new feature. 
    * `chore: set up changelog generation tools`
       Use `chore` to indicate that this PR makes some changes that users don't need to know.
    * `test: fix CI failure. Part of #1234`
        Use `part of` to indicate that a PR is working on an issue, but shouldn't close the issue when merged.
-->

hbelmiro reviewed (commented):
In the link, only `ubuntu-20.04` is mentioned. Have you confirmed that `ubuntu-24.04` also has this issue?

droctothorpe commented:
> In the link, only `ubuntu-20.04` is mentioned. Have you confirmed that `ubuntu-24.04` also has this issue?

Yup! If you check out [this output](https://github.com/kubeflow/pipelines/actions/runs/14395018343/job/40369074728?pr=11817) from a similar PR that includes `df -h` steps before and after the `rm` operation...

<img width="130" alt="image" src="https://github.com/user-attachments/assets/37f25efb-761f-4d7b-a97b-88787be9ad88" />

...you can see that 8 GB are freed up:
<img width="424" alt="image" src="https://github.com/user-attachments/assets/d9266c16-1545-4251-b3f2-c9b093a9ea50" />

hbelmiro commented on .github/workflows/kfp-kubernetes-execution-tests.yml:
```diff
@@ -31,6 +31,11 @@ jobs:
         with:
           python-version: '3.9'
 
+      - name: Free up space in /dev/root
```
Do you mind adding a comment with the link here? Something like the following should be enough.

```suggestion
      # This is intended to address disk space issues that have surfaced intermittently during CI - https://github.com/actions/runner-images/issues/2840#issuecomment-1284059930
      - name: Free up space in /dev/root
```

hbelmiro commented on .github/workflows/sdk-execution.yml:
```diff
@@ -30,6 +30,11 @@ jobs:
         with:
           python-version: 3.9
 
+      - name: Free up space in /dev/root
```
Same as above:

```suggestion
      # This is intended to address disk space issues that have surfaced intermittently during CI - https://github.com/actions/runner-images/issues/2840#issuecomment-1284059930
      - name: Free up space in /dev/root
```

droctothorpe commented:
Addressed your feedback @hbelmiro and added logging to the workflow to log before and after disk utilization.
<img width="465" alt="image" src="https://github.com/user-attachments/assets/de412233-1c37-46e8-a94f-f67599908cde" />

hbelmiro reviewed (approved):
Awesome!
Thank you @droctothorpe.

/lgtm
